readthedocs/proxito/views/hosting.py

```diff
@@ -350,13 +373,20 @@ def _v1(self, project, version, build, filename, url, user):
         main_project = project.main_language_project or project
 
         # Exclude the current project since we don't want to return itself as a translation
-        project_translations = main_project.translations.all().exclude(
-            slug=project.slug
+        project_translations = (
+            Project.objects.public(user=user)
+            .filter(pk__in=main_project.translations.all())
+            .exclude(slug=project.slug)
         )
+
         # Include main project as translation if the current project is one of the translations
         if project != main_project:
-            project_translations |= Project.objects.filter(slug=main_project.slug)
-        project_translations = project_translations.order_by("language")
+            project_translations |= Project.objects.public(user=user).filter(
+                slug=main_project.slug
+            )
+        project_translations = project_translations.order_by("language").select_related(
+            "main_language_project"
+        )
```

Comment:
Why do we need the `main_language_project` as select related here? Where is it used?

Reply:
It's used in the serializer, we are doing lots of n+1 queries there.